pkg/handlers/middleware.go

```diff
@@ -21,23 +35,46 @@ func CorsMiddleware(next http.Handler) http.Handler {
 	})
 }
 
+func DebugLoggingMiddleware(next http.Handler) http.Handler {
```

Comment:
Made this separate for endpoints like healthz, so errors are never logged unless `DEBUG` is enabled.